data/core/tokenizer.lua

```diff
@@ -72,7 +73,7 @@ local function retrieve_syntax_state(incoming_syntax, state)
     -- syntax we're using. Rather than walking the bytes, and calling into
     -- `syntax` each time, we could probably cache this in a single table.
     for i = 0, 2 do
-      local target = bit32.extract(state, i*8, 8)
+      local target = bit.extract(state, i*8, 8)
```

Comment:
The calls to bit32.extract and bit32.replace should probably be replaced with more native logic.

I'm not fluent with Lua so its probably best I leave this to someone else.